container-entrypoints/build-ruby

```diff
@@ -86,23 +86,29 @@ if [[ "$VARIANT" = jemalloc ]]; then
     EXTRA_CONFIGURE_ARGS=(--with-jemalloc)
     if [[ "$PACKAGE_VERSION" =~ ^2\. || "$PACKAGE_VERSION" =~ ^3\.0 ]]; then
         run patch -p1 -ti /system/resources/ruby_26_jemalloc.patch
-    else
+    elif [[ "$PACKAGE_VERSION" =~ ^3\.1 ]]; then
         run patch -p1 -ti /system/resources/ruby_31_jemalloc.patch
+    else
+        EXTRA_CONFIGURE_ARGS+=(--enable-yjit)
+        run patch -p1 -ti /system/resources/ruby_32_jemalloc.patch
     fi
 elif [[ "$VARIANT" = malloctrim ]]; then
     EXTRA_CONFIGURE_ARGS=()
     run patch -p1 -ti /system/resources/ruby_malloctrim.patch
     if [[ "$PACKAGE_VERSION" =~ ^2\. || "$PACKAGE_VERSION" =~ ^3\.0 ]]; then
         run patch -p1 -ti /system/resources/ruby_26_malloctrim.patch
-    else
+    elif [[ "$PACKAGE_VERSION" =~ ^3\.1 ]]; then
         run patch -p1 -ti /system/resources/ruby_31_malloctrim.patch
+    else
+        EXTRA_CONFIGURE_ARGS+=(--enable-yjit)
```

Comment:
It looks like YJIT requires Rust 2021 edition support. YJIT fails to build on the following distributions:

 * debian-9 (rustc 1.41.1)
 * debian-10 (rustc 1.16.0)
 * debian-11 (rustc 1.48.0)

Surprisingly it works on Ubuntu 18.04 (younger than Debian 10) because it has rustc 1.61.0. I tested locally before on Ubuntu 18.04 and compiling YJIT works so I assumed it would also work on Debian 10 and maybe even 9, but apparently not. :(

This is the error:

~~~
building Rust YJIT (release mode)
error: argument for `--edition` must be one of: 2015|2018. (instead was `2021`)
~~~

We have a couple of choices here:

 * Disable YJIT on certain distributions. If we do this then we'll also have to modify the tests to test for YJIT only on distributions where it's not disabled.
 * Install an up-to-date enough Rust in the debian-9 through 11 images. (Hopefully there are binaries...)

What do you think?

Reply:
I've pushed a commit to your branch that implements the latter strategy. It looks like rustup-init indeed installs binaries. Could you let me know what you think?

Reply:
Sorry the delay but looks awesome @FooBarWidget! Thanks!

Glad to see you're able to sort the debian ones.

It looks we have a build passing now :) 🚀